internal/worker/modelworkermanager/modelworkermanager.go

```diff
@@ -226,8 +226,10 @@ func (m *modelWorkerManager) loop() error {
 }
 
 func (m *modelWorkerManager) modelChanged(modelUUID string) error {
-	model, release, err := m.config.Controller.Model(modelUUID)
-	if errors.Is(err, errors.NotFound) {
+	ctx := context.Background()
+	model, err := m.config.ControllerDomainServices.Model().Model(ctx, model.UUID(modelUUID))
```

Comment:
Don't pass in the entire domain services factory. Just declare an interface with a Model(context, uuid) method and pass in ControllerDomainServices.Model(). Then you can easily mock for tests.

Reply:
Sure thanks ian :)